src/ccall/src/goat_game.rs

```diff
@@ -0,0 +1,73 @@
+use amethyst::ecs::prelude::{Component, DenseVecStorage};
+use amethyst::{
+    assets::{Handle, Format, Loader, ProgressCounter},
+    core::{math::Vector3, Transform},
+    prelude::*,
+    renderer::{
+        formats::mesh::ObjFormat,
+        camera::{Camera},
+        types::{Mesh, MeshData},
+    },
+};
+
+use crate::goat::{*};
+
+pub struct GoatGame;
+pub struct Model {}
+
+impl Component for Model {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl SimpleState for GoatGame {
+    fn on_start(&mut self, data: StateData<'_, GameData<'_, '_>>) {
+        let StateData { world, .. } = data;
+
+        initialise_camera(world);
+        initialize_model(world);
+    }
+}
+
+fn initialise_camera(world: &mut World) {
+    let mut transform = Transform::default();
+    transform.set_translation_xyz(0.0, 12.0, -5.0);
+
+    world
+        .create_entity()
+        .with(Camera::standard_3d(100.0, 100.0))
+        .with(transform)
+        .build();
+}
+
+fn initialize_model(world: &mut World) {
+    let asset : Handle<Mesh> = {
+        let loader = world.read_resource::<Loader>();
+        let mut progress = ProgressCounter::default();
+        let mesh_storage = world.read_resource();
+        rs_goat_init();
+        println!("generating goat");
+        let g = Goat::random();
+        println!("printing before mesh");
+        let m = g.mesh();
+        let (v, _f) = m.buffers();
+        println!("printing before vertices {:?}", v.to_vec());
+        let mesh_data : MeshData = ObjFormat.import_simple(v.to_vec()).unwrap();
```

Comment:
`v` isn't in the right format yet. It looks like we might be able to use 
https://docs.amethyst.rs/v0.10.0/amethyst_renderer/struct.MeshBuilder.html
e.g. first comment here https://www.reddit.com/r/rust_gamedev/comments/c3s9wi/need_help_creating_a_mesh_programmatically_with/

Unfortunately, meshbuilder doesn't support index lists 🤦‍♂ Lets get it working with trianglestrip first. Then I'll modify the library to output vertices formatted as a trianglestrip.

Reply:
does `to_vec()` make a copy of the slice?

Reply:
oh my mistake, it looks like you can load indices using with_indices as in the example in reddit thread

Reply:
so faces are just the triangle indices

Reply:
I updated to try to use `with_indices` but it might still not be working 😬